### PR TITLE
ITP_JAN_25 | HASAN DEMIROZ | Module-Structuring-and-Testing-Data | Sprint 3

### PR DESCRIPTION
**Pull request by hasandemiroz**
## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: Sprint-3/3-mandatory-practice/implement/count.test.js [hunk 1]
```diff
@@ -22,3 +22,25 @@ test("should count multiple occurrences of a character", () => {
...
+  expect(count).toEqual(0);
+});
+
+
+test("should return 0 if character does not appears",() =>{
+  const str = "apple";
+  const char = "c";
+  const count = countChar(str,char);
+  expect(count).toEqual(0);
+});
```
Comment (HappyCoder5114) [inline Sprint-3/3-mandatory-practice/implement/count.test.js:46]: there is a corner case which could potentially failed . when findchar is  empty string "".

**Review by HappyCoder5114 — COMMENTED** [review]
please see the comments.

File: Sprint-3/1-key-implement/2-is-proper-fraction.js [hunk 2]
```diff
@@ -8,7 +8,11 @@
 // write one test at a time, and make it pass, build your solution up methodically
 
 function isProperFraction(numerator, denominator) {
-    if (numerator < denominator) return true;
+    if (Math.abs(numerator) < Math.abs(denominator)) 
+      return true;
+    else if (Math.abs(numerator) > Math.abs(denominator)) return false;
```
[the remark this answers is not in the document]
Comment (hasandemiroz) [reply Sprint-3/1-key-implement/2-is-proper-fraction.js]: I have updated the code and used less line of codes. Thanks.

File: Sprint-3/1-key-implement/2-is-proper-fraction.js [hunk 3]
```diff
@@ -40,14 +44,23 @@ assertEquals(improperFraction, false);
...
 // Input: numerator = 3, denominator = 3
 // target output: false
 // Explanation: The fraction 3/3 is not a proper fraction because the numerator is equal to the denominator. The function should return false.
 const equalFraction = isProperFraction(3, 3);
+assertEquals(equalFraction, false);
 // ====> complete with your assertion
 
 // Stretch:
 // What other scenarios could you test for?
+const numeratorZero = isProperFraction(0, 3);
```
[the remark this answers is not in the document]
Comment (hasandemiroz) [reply Sprint-3/1-key-implement/2-is-proper-fraction.js:60]: The function updated and it returns false for the 0/3 is fraction or not case.

File: Sprint-3/1-key-implement/3-get-card-value.js [hunk 4]
```diff
@@ -8,7 +8,12 @@
 // write one test at a time, and make it pass, build your solution up methodically
 // just make one change at a time -- don't rush -- programmers are deep and careful thinkers
 function getCardValue(card) {
+    let rank = card.slice(0, -1);
     if (rank === "A") return 11;
+    if (["J", "Q", "K", "10"].includes(rank)) return 10;
```
[the remark this answers is not in the document]
Comment (hasandemiroz) [reply Sprint-3/1-key-implement/3-get-card-value.js]: The code has updated. Before the statement it makes it uppercase the it.

File: Sprint-3/3-mandatory-practice/implement/get-ordinal-number.test.js [hunk 5]
```diff
@@ -11,3 +11,18 @@ const getOrdinalNumber = require("./get-ordinal-number");
...
+    expect(getOrdinalNumber(2)).toEqual("2nd");
+    });
+
+
+test("should return '8th' for 8", () => {
+    expect(getOrdinalNumber(8)).toEqual("8th");
+    });
+
+test("should return 'th' for greater than 3", () => {
+    for (let number = 4; number < 20; number++) {
```
[the remark this answers is not in the document]
Comment (hasandemiroz) [reply Sprint-3/3-mandatory-practice/implement/get-ordinal-number.test.js]: The code has updated according to edge cases and works fine now.

File: Sprint-3/3-mandatory-practice/implement/count.test.js (hunk 1 above)
Comment (hasandemiroz) [reply Sprint-3/3-mandatory-practice/implement/count.test.js:53]: It is updated and fixed.

**Comment by hasandemiroz** [comment]
> hey this looks okay,could be merged now. but i dont have write access still. could anyone who who write access preceed this to completed ?

Thank you for your review. I have updated the label as complete now.
